hail/python/hailtop/batch/functions.py

```diff
@@ -0,0 +1,111 @@
+import math
+
+from typing import List
+
+from ..utils.utils import grouped
+from .utils import BatchException
+from .batch import Batch
+from .resource import ResourceGroup, ResourceFile
+
+
+def concatenate(b: Batch, files: List[ResourceFile], branching_factor: int = 100):
+    """
+    Concatenate files using tree aggregation.
+
+    Notes
+    -----
+    Runs the `cat` command aggregating files using the `branching_factor`.
```

Comment:
I think this sentence is just an extended description, I don't think it needs to go under the Notes heading. We also usually put the Notes section after Examples.